From e30463200592a2c099ddfa198105cf2f153a7d70 Mon Sep 17 00:00:00 2001
Message-Id: <e30463200592a2c099ddfa198105cf2f153a7d70.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: Thomas Daniel <thomas.daniel@intel.com>
Date: Mon, 9 Feb 2015 17:22:59 +0000
Subject: [PATCH 59/61] SQUASHME! [VPG]: drm/i915: Fix address in
 logical_ring_write_active_request

logical_ring_write_active_request uses MI_STORE_DWORD_INDEX which takes an index
into the HWSP as its address operand.  However, a full GGTT address is being
written including reserved bits.

Fix this by only using I915_GEM_ACTIVE_SEQNO_INDEX as the address operand.

SQUASHME!: This should be sqaushed with below existing patch
    Author: John Harrison<John.C.Harrison@Intel.com>
    Change-Id: I8c2653787a1b15b4eb00c1dfa05b5e50ab4d2390
    REVERTME [VPG]: drm/i915: Refactored native sync code for better file splits

FOR: ACD-90
Change-Id: I4fca554447cd2a22c77c6b2d2ef30192a9bc69b2
Signed-off-by: Thomas Daniel <thomas.daniel@intel.com>
---
 drivers/gpu/drm/i915/intel_lrc.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_lrc.c b/drivers/gpu/drm/i915/intel_lrc.c
index 6e83d68..afed58e7 100644
--- a/drivers/gpu/drm/i915/intel_lrc.c
+++ b/drivers/gpu/drm/i915/intel_lrc.c
@@ -1249,17 +1249,14 @@ logical_ring_write_active_request(struct intel_ringbuffer *ringbuf,
 				  struct drm_i915_gem_request *req)
 {
 	int ret;
-	struct intel_engine_cs *ring = ringbuf->ring;
 
 	ret = intel_logical_ring_begin(ringbuf, 4);
 	if (ret)
 		return ret;
 
 	intel_logical_ring_emit(ringbuf, MI_STORE_DWORD_INDEX);
-	intel_logical_ring_emit(ringbuf,
-				(ring->status_page.gfx_addr +
-				 (I915_GEM_ACTIVE_SEQNO_INDEX <<
-				  MI_STORE_DWORD_INDEX_SHIFT)));
+	intel_logical_ring_emit(ringbuf, I915_GEM_ACTIVE_SEQNO_INDEX <<
+				  MI_STORE_DWORD_INDEX_SHIFT);
 	intel_logical_ring_emit(ringbuf, i915_gem_request_get_seqno(req));
 	intel_logical_ring_emit(ringbuf, MI_NOOP);
 	intel_logical_ring_advance(ringbuf);
-- 
1.7.9.5

